pandas/tests/tools/test_to_datetime.py

```diff
@@ -1076,31 +1076,39 @@ def test_to_datetime_array_of_dt64s(self, cache, unit):
         # Assuming all datetimes are in bounds, to_datetime() returns
         # an array that is equal to Timestamp() parsing
         result = to_datetime(dts, cache=cache)
-        expected = DatetimeIndex([Timestamp(x).asm8 for x in dts], dtype="M8[ns]")
+        if cache:
+            # FIXME: behavior should not depend on cache
+            expected = DatetimeIndex([Timestamp(x).asm8 for x in dts], dtype="M8[s]")
+        else:
+            expected = DatetimeIndex([Timestamp(x).asm8 for x in dts], dtype="M8[ns]")
```

Comment:
Do you have a inking what code path is casting the result to ns?

Would be good to have an issue for this once it's merged

Reply:
array_to_datetime/maybe_convert_objects.  there's a checkbox for each in #46587